storage/src/main/java/com/tencent/rss/storage/handler/impl/MultiStorageHdfsShuffleReadHandler.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Tencent is pleased to support the open source community by making
+ * Firestorm-Spark remote shuffle server available.
+ *
+ * Copyright (C) 2021 THL A29 Limited, a Tencent company.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * https://opensource.org/licenses/Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.tencent.rss.storage.handler.impl;
+
+import com.tencent.rss.common.ShuffleDataResult;
+import com.tencent.rss.common.ShuffleDataSegment;
+import com.tencent.rss.common.ShuffleIndexResult;
+import com.tencent.rss.common.util.RssUtils;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.hadoop.conf.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MultiStorageHdfsShuffleReadHandler extends HdfsShuffleReadHandler {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MultiStorageHdfsShuffleReadHandler.class);
+
+  private final int partitionId;
+  private long dataFileOffset;
+
+  public MultiStorageHdfsShuffleReadHandler(
+      int partitionId,
+      String filePrefix,
+      int readBufferSize,
+      Configuration conf)  throws IOException {
+    super(filePrefix, readBufferSize, conf);
+    this.partitionId = partitionId;
+  }
+
+  public ShuffleDataResult readShuffleData() {
```

Comment:
Can we reuse readShuffleData with HdfsShuffleReadHandler?

Reply:
Yes, this method is already moved by extends HdfsShuffleReadHandler.

Reply:
The commit somehow missed. I've pushed it. We only overwrite the special part of multi handler.